### PR TITLE
Adding Unit Tests for Widget Meta Dao and Fixing the API issues.

### PR DESCRIPTION
**Pull request by ksdperera**

### CODE REVIEW

File: components/dashboards/org.wso2.carbon.dashboards.core/src/main/java/org/wso2/carbon/dashboards/core/WidgetMetadataProvider.java [hunk 1]
```diff
@@ -58,6 +58,9 @@
      */
     Set<WidgetMetaInfo> getAllWidgetConfigurations() throws DashboardException;
 
+
+    void delete(String widgetId) throws DashboardException;
```
Comment (this) [inline components/dashboards/org.wso2.carbon.dashboards.core/src/main/java/org/wso2/carbon/dashboards/core/WidgetMetadataProvider.java]: Please add javadoc for this method

File: components/dashboards/org.wso2.carbon.dashboards.api/src/main/java/org/wso2/carbon/dashboards/api/internal/WidgetRestApi.java [hunk 2]
```diff
@@ -158,19 +160,41 @@ private static Response serverErrorResponse(String message) {
...
+     */
+    @DELETE
+    @Path("/{id}")
+    public Response deleteWidget(@PathParam("id") String widgetId) {
+        try {
+            if (widgetMetadataProvider.isWidgetPresent(widgetId)) {
+                widgetMetadataProvider.delete(widgetId);
+                return Response.status(OK).build();
+            } else {
+                return Response.status(NO_CONTENT).entity("Cannot find widget '" + widgetId + "'.").build();
```
Comment (this) [inline components/dashboards/org.wso2.carbon.dashboards.api/src/main/java/org/wso2/carbon/dashboards/api/internal/WidgetRestApi.java]: Incorrect HTTP status code. Instead use `UNPROCESSABLE_ENTITY` https://httpstatuses.com/422

File: components/dashboards/org.wso2.carbon.dashboards.core/src/main/java/org/wso2/carbon/dashboards/core/WidgetMetadataProvider.java (hunk 1 above)
Comment (ksdperera) [reply components/dashboards/org.wso2.carbon.dashboards.core/src/main/java/org/wso2/carbon/dashboards/core/WidgetMetadataProvider.java]: Fixed by: c7f9a9abd89b29901f03eaf82fab6dbf58f3ad8f

File: components/dashboards/org.wso2.carbon.dashboards.api/src/main/java/org/wso2/carbon/dashboards/api/internal/WidgetRestApi.java (hunk 2 above)
Comment (ksdperera) [reply components/dashboards/org.wso2.carbon.dashboards.api/src/main/java/org/wso2/carbon/dashboards/api/internal/WidgetRestApi.java]: Fixed by: e809f3a84c17eb9b310c7c778f20eff7d10c9c79

**Pull request merged**